docs/tutorial/streamer_design.md

```diff
@@ -1,29 +1,398 @@
+}
+
+data_writer_params:{
+  tcdm_ports_num: [8],
+  spatial_bounds: [[8]],
+  spatial_dim: [1],
+  element_width: [64],
+}
+```
+
+The number of elements in a list pertain to how many data movers there are. For example, there are 2 elements in the `data_reader_params` and therefore it instantiates two read data movers. There is only 1 element in the `data_writer_params` and hence only instantiates 1 write data mover.
+
+Finally there is the `stationarity` configuration which is for stationarity for each data mover. If the stationarity bit is set, the innermost loop for that data mover is set to 1. This does not instantiate anything but rather affects the loop bound for a data mover. Basically, it only fixes the loop bound to 1. This is a special case scenario only. In the SNAX ALU, this is 0. 
+
+## SNAX ALU Streamer CSRs
+
+Based on the configuration file we expect the CSR registers to have:
+
+- 1 register for a single loop bound as specified in the `temporal_addrgen_unit_params -> loop_dim = 1`. If the number is 2 then we would have 2 loop bounds.
+- 3 temporal stride registers, 3 spatial stride registers, and 3 base addres spointers since we have 3 data movers in total (2 read data movers and 1 write data movers).
+- 1 start register to start the streamer.
+- 1 performance counte register for how many cycles the streamer was active.
+
+Below is a tabulated version of the CSRs of the streamers with the register address, type, and description:
+
+|  register name      |  register addr  |   type  |                   description                        |
+| :-----------------: | :-------------: | :-----: |:---------------------------------------------------- |
+| temporal loop bound |       0         |   RW    | Temporal loop bound for address generation           |
+| temporal stride 0   |       1         |   RW    | Temporal stride for input A                          |
+| temporal stride 1   |       2         |   RW    | Temporal stride for input B                          |
+| temporal stride 2   |       3         |   RW    | Temporal stride for output OUT                       |
+| spatial stride 0    |       4         |   RW    | Spatial stride for input A                           |
+| spatial stride 1    |       5         |   RW    | Spatial stride for input B                           |     
+| spatial stride 2    |       6         |   RW    | Spatial stride for output OUT                        |     
+| base addr 0         |       7         |   RW    | Base address for input A                             |
+| base addr 1         |       8         |   RW    | Base address for input A                             |
+| base addr 2         |       9         |   RW    | Base address for output OUT                          |
+| start               |       10        |   RW    | Start register to send configurations                |
+| perf. counter       |       11        |   RO    | Performance counter indicating number of cycles ran  | 
```

Comment:
> Performance counter indicating number of cycles ran

How is this different from `mcycle`?

Reply:
`mcycle` only tags the current cycle count from boot up. So technically it just measures the current cycle time from boot up.

performance counters are also more accurate because they directly measures the "busy" or "active" time of an accelerator. If we just do the cycle tagging time from the traces, we also capture cycles caused by setup time like branch conditions or such.